### ISSUE TITLE
[flat config] `ecmaVersion: "latest"`

### ISSUE THREAD

**Issue by mdjermanovic** [eslint/eslint#15366]
### ESLint version

`flat-config-linter` branch

### What problem do you want to solve?

ref: https://github.com/eslint/eslint/pull/15185#discussion_r757536540

With this configuration, ESLint will use `espree.latestEcmaVersion` as ecmaVersion:

```js
{
    languageOptions: {
        ecmaVersion: "latest"
    }
}
```

### What do you think is the correct solution?

ESLint should maintain its own `latestEcmaVersion` instead of using `espree.latestEcmaVersion`.

`languageOptions.ecmaVersion` is an ESLint option, and it is used not only for parsing.

An in-range version of `espree` can support `ecmaVersion` that is not supported by currently used version of ESLint. If we use `espree.latestEcmaVersion`, parsing will work well, but other features that are based on `ecmaVersion` may not. For example, in the mentioned scenario, ESLint will not add any ES globals

### Participation

- [X] I am willing to submit a pull request for this change.

**Comment by btmills** [comment]
Good call. Is there any discussion we need to have here before marking as accepted and moving to Ready to Implement?

**Comment by nzakas** [comment]
Good idea. 👍

**Comment by fasttime** [comment]
@mdjermanovic Would you like to submit a PR for this change?
